worlds/sc2/Rules.py

```diff
@@ -607,12 +596,10 @@ def protoss_competent_comp(self, state: CollectionState) -> bool:
 
     def protoss_stalker_upgrade(self, state: CollectionState) -> bool:
         return (
-                state.has_any(
-                    {
-                        ItemNames.STALKER_INSTIGATOR_SLAYER_DISINTEGRATING_PARTICLES,
-                        ItemNames.STALKER_INSTIGATOR_SLAYER_PARTICLE_REFLECTION
-                    }, self.player)
-                and self.lock_any_item(state, {ItemNames.STALKER, ItemNames.INSTIGATOR, ItemNames.SLAYER})
+            state.has_any({
+                ItemNames.STALKER_INSTIGATOR_SLAYER_DISINTEGRATING_PARTICLES,
+                ItemNames.STALKER_INSTIGATOR_SLAYER_PARTICLE_REFLECTION
+            }, self.player)
```

Comment:
Won't this following code remove the required items and fail the generation:
```python
        # LotV
        # Shared unit upgrades between several units
        if not {ItemNames.STALKER, ItemNames.INSTIGATOR, ItemNames.SLAYER} & logical_inventory_set:
            inventory = [item for item in inventory if not item.name.endswith("(Stalker/Instigator/Slayer)")]
            unused_items = [item_name for item_name in unused_items if not item_name.endswith("(Stalker/Instigator/Slayer)")]
```

Reply:
You're right. The Evil Awoken logic is a real pain for generation; I'm asking in the discord if we should rework it somehow. If we keep it, I think the solution is to add the `Locked` flag in `__init__.py` when adding the AllowedOrphan flags. The alternatives are to loosen the logic or to _also_ refactor PoolFilter.py, which would make this already-large PR even larger.

Will add a unit test for this as well.

Reply:
I think the solution is to:
While parsing excluded items in `ValidInventory.generate_reduced_inventory()`, first process excluded flags:
shuffle items with excluded flag ans validate against logic. If logic fails, the item gets locked instead and a warning is emitted
Then regular size-based filtering would happen
You also need to change that method in order to `allowOrphan` work correctly (and the rule dropping Stalker upgrades should take it into account)

Using this approach would also keep the logic able to handle standard tactics on future Mastery locations without making the game too rough

This rule is written in a way so the upgrades won't get orphaned thus the item filtering won't remove them. The proper handling of allowOrphan flag should solve that problem instead.

Reply:
I see where you're coming from implementation-wise, but I'm saying it's a bad requirement. If a player selects an option saying "only include vanilla items", and they got non-vanilla items, that's a bug, no two ways around it. When faced with a situation where it must be possible to generate this mission with vanilla items only, and where this mission requires non-vanilla items, one requirement must go. This is why I took this to the discord, and the users agreed that the logic requirement should be removed.

I don't want to refactor all of PoolFilter.py as well, as this PR is already colossal, and I think there are more easy improvements to be made by touching up mission selection / mission order generation / adding mission groups. If I was to refactor it, I would pass in the complete item list with flags, and use those throughout the file rather than the current set-inclusion based filtering. Work for a future PR, but I'd like to limit the scope at least somewhat.

Reply:
Clean it up (it gets unused) and mark those items as useful (they won't be logic items anymore).